test/hotspot/jtreg/runtime/ErrorHandling/TestGZippedHeapDumpOnOutOfMemoryError.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2021 SAP SE. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary Test verifies that -XX:HeapDumpGzipLevel=0 works
+ * @library /test/lib
+ * @run driver/timeout=240 TestGZippedHeapDumpOnOutOfMemoryError run 0
+ */
+
+/*
+ * @test
+ * @summary Test verifies that -XX:HeapDumpGzipLevel=1 works
+ * @library /test/lib
+ * @run driver/timeout=240 TestGZippedHeapDumpOnOutOfMemoryError run 1
+ */
```

Comment:
If the tests are fast enough, I would add at least another one at the end of the valid gzip range.

Reply:
I don't think that this is needed, since these test should not test the gzipping (there are already tests for that).